components/kyma-environment-broker/internal/process/provisioning/conn_bind.go

```diff
@@ -0,0 +1,187 @@
+		return s.handleError(operation, fmt.Errorf("Connectivity Provisioning step was not triggered"), log, "")
+	}
+
+	smCli, err := operation.ServiceManagerClient(log)
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("unable to create Service Manage client"))
+	}
+	// test if the provisioning is finished, if not, retry after 10s
+	resp, err := smCli.LastInstanceOperation(operation.Conn.Instance.InstanceKey(), "")
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("LastInstanceOperation() call failed"))
+	}
+	log.Infof("Provisioning Connectivity (instanceID=%s) state: %s", operation.Conn.Instance.InstanceID, resp.State)
+	switch resp.State {
+	case servicemanager.InProgress:
+		return operation, 10 * time.Second, nil
+	case servicemanager.Failed:
+		return s.operationManager.OperationFailed(operation, fmt.Sprintf("Connectivity provisioning failed: %s", resp.Description), log)
+	}
+	// execute binding
+	var connectivityOverrides *ConnectivityOverrides
+	if !operation.Conn.Instance.Provisioned {
+		if operation.Conn.BindingID == "" {
+			operation.Conn.BindingID = uuid.New().String()
+		}
+		respBinding, err := smCli.Bind(operation.Conn.Instance.InstanceKey(), operation.Conn.BindingID, nil, false)
+		if err != nil {
+			return s.handleError(operation, err, log, fmt.Sprintf("Bind() call failed"))
+		}
+		// get overrides
+		connectivityOverrides, err = GetConnCredentials(respBinding.Binding)
+		if err != nil {
+			return s.handleError(operation, err, log, fmt.Sprintf("getCredentials() call failed"))
+		}
+		encryptedOverrides, err := EncryptConnOverrides(s.secretKey, connectivityOverrides)
+		if err != nil {
+			return s.handleError(operation, err, log, fmt.Sprintf("encryptOverrides() call failed"))
+		}
+
+		// save the status
+		op, retry := s.operationManager.UpdateOperation(operation, func(operation *internal.ProvisioningOperation) {
+			operation.Conn.Overrides = encryptedOverrides
+			operation.Conn.Instance.Provisioned = true
+			operation.Conn.Instance.ProvisioningTriggered = false
+		}, log)
+		if retry > 0 {
+			log.Errorf("unable to update operation")
+			return operation, time.Second, nil
+		}
+		operation = op
+	} else {
+		// get the credentials from encrypted string in operation.Conn.Instance.
+		connectivityOverrides, err = DecryptConnOverrides(s.secretKey, operation.Conn.Overrides)
+		if err != nil {
+			return s.handleError(operation, err, log, fmt.Sprintf("decryptOverrides() call failed"))
+		}
+	}
+
+	// TODO: Decide how we want to pass this data to the SKR. Currently,
+	//       credentials are prepared as a ConnectivityOverrides structure.
+	//       See the github card - https://github.com/orgs/kyma-project/projects/6#card-56776111
+	//       ...
+	//       - [ ] define what changes need to be done in KEB to
+	//             allow passing secrets data to the Provisioner
+	log.Debugf("Got Connectivity Service credentials from the binding: %+v", connectivityOverrides)
```

Comment:
keep in mind that such overrides contains sensitive data and should not be logged.

Reply:
Ok! It's just for the debugging reason here. I'll comment it then before passing the PR to the review